ledger/ledger-api-client/src/main/scala/com/digitalasset/ledger/client/services/commands/tracker/CommandTracker.scala

```diff
@@ -225,10 +226,19 @@ private[commands] class CommandTracker[Context](maxDeduplicationTime: () => JDur
             }
             val commandTimeout = {
               lazy val maxDedup = maxDeduplicationTime()
-              val dedup = commands.deduplicationTime.getOrElse(
-                ProtoDuration.of(maxDedup.getSeconds, maxDedup.getNano)
-              )
-              Instant.now().plusSeconds(dedup.seconds).plusNanos(dedup.nanos.toLong)
+              lazy val protoMaxDedup = ProtoDuration.of(maxDedup.getSeconds, maxDedup.getNano)
+              val timeoutDuration = commands.deduplication match {
+                case Deduplication.Empty =>
+                  protoMaxDedup
+                case Deduplication.DeduplicationTime(value) =>
+                  value
+                case Deduplication.DeduplicationStart(_) =>
+                  protoMaxDedup
```

Comment:
If we're going to ignore this, can we make sure the work is captured somewhere? And add a big fat comment explaining this?

Reply:
There is a story to remove this code all together and should be done soon. Do you think I should comment with the issue id here?
I decided to not calculate the duration because I would not want to have to handle `maxSkew` as well here, and it's just safer to use max duration.